pandas/core/generic.py

```diff
@@ -9953,11 +9953,24 @@ def _check_percentile(self, q):
         """
 
     @Appender(_shared_docs['pct_change'] % _shared_doc_kwargs)
-    def pct_change(self, periods=1, fill_method='pad', limit=None, freq=None,
-                   **kwargs):
+    def pct_change(self, periods=1, fill_method=None, limit=None, freq=None,
+                   skipna=None, **kwargs):
+        if skipna and fill_method is not None:
+            raise ValueError("cannot pass both skipna and fill_method")
+        elif skipna and limit is not None:
+            raise ValueError("cannot pass both skipna and limit")
+        if skipna is None and fill_method is None and limit is None:
+            skipna = True
+        if skipna and self._typ == 'dataframe':
```

Comment:
use isinstance here

Reply:
Hmm maybe add this to the frame subclass instead? Somewhat confusing to introspect here in the shared one.

Reply:
For the moment, I have added `isinstance` as required by @jreback. Tell me if you both think I should do otherwise.